### PR TITLE
Throw validation error on route conflicts (Subo #53)

### PR DESCRIPTION
**Pull request by rnpridgeon**
intended to close https://github.com/suborbital/subo/issues/53

### CODE REVIEW

File: directive/directive_validator.go [hunk 1]
```diff
@@ -127,6 +128,16 @@ func (d *Directive) Validate() error {
 			if h.RespondTo != "" {
 				problems.add(fmt.Errorf("handler for resource %s has type 'request', but defines a 'respondTo' field, which only valid for type 'stream'", h.Input.Resource))
 			}
+
+			if resource, ok := inputMap[h.Input.Resource]; ok {
+				if _, ok := resource[h.Input.Method]; ok {
+					problems.add(fmt.Errorf("duplicate entry found for resource %s method %s", h.Input.Resource, h.Input.Method))
```
[the remark this answers is not in the document]
Comment (rnpridgeon) [reply directive/directive_validator.go:134]: That's fair I wasn't considering parameterized segments. Will adjust to leverage vektor mux as instructed in the original issue

**Review by cohix — APPROVED** [review]
Wonderful. I took a look at the code and It makes sense, someone else should test it out before we merge it.

**Comment by rnpridgeon** [comment]
Manual testing revealed subo doesn't actually verify the directive. The following PR fixes this 

https://github.com/suborbital/subo/pull/208

**Comment by rnpridgeon** [comment]
Had to tweak this a bit to use a non-nil handler implementation in order to catch conflicts in constant paths. 

I have tested this pr with https://github.com/suborbital/subo/pull/208 and it appears to work as expected.

**Pull request merged**